createBookFromReadme.sh

```diff
@@ -6,7 +6,7 @@
 # -------------------- Utility Methods --------------------
 # Check for binaries
 function checkEnvironment(){
-    type gcsplit >/dev/null 2>&1 || { echo "Install 'gcsplit' first (e.g. via 'brew install coreutils')." >&2 && exit 1 ; }
+    type csplit >/dev/null 2>&1 || { echo "Install gcsplit' first (e.g. via 'brew install coreutils')." >&2 && exit 1 ; }
```

Comment:
I wouldn't remove the `gcsplit` line. 

Because on macOS gcsplit is used and it cannot be caught in Environment. 

What you want to do is check for os type here using an if-else and then execute the check line for that os type.